amtl/am-string.h

```diff
@@ -140,6 +156,14 @@ class AString
     chars_[length] = '\0';
   }
 
+  static AString concat(const char *a, size_t alen, const char *b, size_t blen) {
+    AutoArray<char> buf;
+    buf = new char[alen + blen + 1];
+    strcpy(buf, a);
+    strcat(buf, b);
```

Comment:
It'd be more efficient to `strcpy(&buf[alen], b);` instead of `strcat`.

Thinking more on overflow - maybe we should just call `abort()`. It's the same as we do for an OOM right now. And we could wrap the two Try methods above in a `TryUintptrAdd` method.